### PR TITLE
Added Active Learning Example

### PR DESCRIPTION
**Pull request by DarshanDeshpande**
This example demonstrates ratio-based sampling for active learning on the IMDB reviews dataset.
Features of this example:
1. Discusses the applicability of Active Learning for businesses, especially with the rising trend in data-centric AI.
2. Uses active learning to train a small model on 30,000 samples, yielding comparable results to the same model trained on the full dataset containing 40,000 training samples.
3. Shows that ratio-based sampling can significantly balance out false positives and negatives which makes it more suitable in deployment scenarios where both labels (for binary classification) must be given equal importance.

Colab notebook for this example: https://colab.research.google.com/drive/1lUNoXWeHPwla8ISdKhhub4lpQv4TCdp0?usp=sharing

@fchollet Please let me know if there are any changes you would like me to make.
Thanks!

### CODE REVIEW

File: examples/nlp/active_learning_review_classification.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+    dataset_len = train_samples * 2
+
+    model = create_model()
+    # We will monitor the false positives and false negatives predicted by our model
+    # These will decide the subsequent sampling ratio for every active learning loop
+    model.compile(
+        loss="binary_crossentropy",
+        optimizer="rmsprop",
+        metrics=[
+            "accuracy",
```
Comment (fchollet) [inline examples/nlp/active_learning_review_classification.py:252]: For consistency, avoid mixing strings and objects: use objects everywhere

File: examples/nlp/active_learning_review_classification.py [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+
+"""
+## Creating the Model
+"""
+
+
+def create_model():
+    model = keras.models.Sequential()
+    model.add(layers.Embedding(3000, 128))
+    model.add(layers.Bidirectional(layers.LSTM(32, return_sequences=True)))
```
Comment (fchollet) [inline examples/nlp/active_learning_review_classification.py:155]: You might want to use a bigram model instead, which is what works best on IMDB, since it will be much faster.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! This is certainly a valuable topic to cover!

**Comment by DarshanDeshpande** [comment]
@fchollet I have updated the example. I couldn't find a major training time difference in bigram and unigram models so I stuck with this one. The goal of this example is to demonstrate the techniques and not to build the best model so I think this model should suffice (~10s/epoch when using `cache()`). Additionally, let me know if the compile argument must have all objects or is it okay if only metrics are passed as objects?

If there is anything else you want me to change then please let me know!

File: examples/nlp/active_learning_review_classification.py [hunk 3]
```diff
@@ -0,0 +1,519 @@
...
+ensured that the model architecutre is capable of overfitting the initial data. This
+overfitting gives a strong hint that the model will have enough capacity to fit to
+future, unseen data.
+"""
+
+
+def create_model():
+    model = keras.models.Sequential(
+        [
+            layers.Embedding(3000, 128),
```
Comment (fchollet) [inline examples/nlp/active_learning_review_classification.py:248]: Use explicit keyword arguments here to improve readability

File: examples/nlp/active_learning_review_classification.py [hunk 4]
```diff
@@ -0,0 +1,519 @@
...
+We create a small, bidirectional LSTM model. When using active learning, it must be
+ensured that the model architecutre is capable of overfitting the initial data. This
+overfitting gives a strong hint that the model will have enough capacity to fit to
+future, unseen data.
+"""
+
+
+def create_model():
+    model = keras.models.Sequential(
+        [
```
Comment (fchollet) [inline examples/nlp/active_learning_review_classification.py:247]: Add an `Input()`, also to improve readability

File: examples/nlp/active_learning_review_classification.py [hunk 5]
```diff
@@ -0,0 +1,519 @@
...
+    )
+    model.summary()
+    return model
+
+
+"""
+## Training on the entire dataset
+
+To show the effectiveness of Active Learning, we will first train the model on the entire
+dataset containing 40,000 labeled samples. This model will be used for comparison later.
```
Comment (fchollet) [inline examples/nlp/active_learning_review_classification.py:264]: You use both the spelling "labell*" and "label*", pic one

File: examples/nlp/active_learning_review_classification.py [hunk 6]
```diff
@@ -0,0 +1,519 @@
...
+    .cache()
+    .shuffle(20000)
+)
+
+# Training the full model
+full_dataset_model = train_full_model(full_train_dataset, val_dataset, test_dataset)
+
+"""
+## Training via Active Learning
+A general pipeline that we follow when performing active learning is demonstrated below:
```
Comment (fchollet) [inline examples/nlp/active_learning_review_classification.py]: You spell "active learning" both with capital letters and without, pick one

**Review by fchollet — COMMENTED** [review]
Looks great! I pushed some copyedits, please pull them first. Some last comments to address, then we should be ready.

**Comment by DarshanDeshpande** [comment]
@fchollet I have made the final changes. Please have a look.

**Review by fchollet — APPROVED** [review]
Awesome, thanks! You can add the generated files 👍

**Comment by DarshanDeshpande** [comment]
@fchollet I have added the generated files. Please check.

**Comment by fchollet** [comment]
Thanks for the great contribution!
